src/shared/lib/validation/file.ts

```diff
@@ -19,8 +23,14 @@ export function validateFileName(name: string, validExtensions: string[] = []):
   }
 }
 
-export function validateFilePath(path: string): Validation<string> {
-  return validateGenericString(path, 'File path');
+export function validateFilePermissions(raw: any): ArrayValidation<FilePermissions<Id, UserType>> {
+  raw = isArray(raw) ? raw : [raw];
+  const validatedFilePermissions = validateArray(raw, validateFilePermission);
+  return mapValid(validatedFilePermissions, perms => {
+    return Array
+      .from(new Set(perms.map(v => JSON.stringify(v))))
+      .map(v => JSON.parse(v) as FilePermissions<Id, UserType>);
+  });
 }
```

Comment:
@sutherlanda I refactored this function to take advantage of validation's helper functions (i.e. mapValid). The semantic change was that non-array raw values are coerced into a single-element array.